### PR TITLE
Google ads v11 to v12

### PR DESCRIPTION
**Pull request by immanojkumar**
This PR is to Upgrade Google enhanced conversations action destination from version v11 to v12 with flagon feature. 

Jira: [Jira Epic](https://segment.atlassian.net/browse/STRATCONN-1779)
Dropbox Document: [Upgrade Google Ads API from v11 to v12 ](https://paper.dropbox.com/doc/Upgrade-Google-Ads-API-from-v11-to-v12--Bx9VNl48Vlr_e7RnOa24ElERAg-RZpy1xjodNZ9jfNUQsuH9)
Flag:  [google-enhanced-conversions-v12](https://flagon.segment.com/families/centrifuge-destinations/gates/google-enhanced-v12)

## Testing
Detailed test document is available here
https://docs.google.com/document/d/1ojsDL-Sei_Tez6rENHnIOBpBbvQwmJpxB7Fb53THdkM/edit?usp=share_link

- [x] Added unit tests.
- [x] Tested in the staging environment

### CODE REVIEW

**Comment by sayan-das-in** [comment]
@immanojkumar can you change the PR description to match the original template. Could you also link some screenshots / test documents ?

File: packages/destination-actions/src/destinations/google-enhanced-conversions/__tests__/uploadCallConversion.test.ts [hunk 1]
```diff
@@ -95,6 +95,61 @@ describe('GoogleEnhancedConversions', () => {
       expect(responses[1].status).toBe(201)
     })
 
+    it('maps custom variables correctly', async () => {
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/__tests__/uploadCallConversion.test.ts:98]: should we give a different name to this test? Like `uses v12 when google-enhanced-v12 flag is enabled`

File: packages/destination-actions/src/destinations/google-enhanced-conversions/__tests__/uploadClickConversion.test.ts [hunk 2]
```diff
@@ -146,6 +146,62 @@ describe('GoogleEnhancedConversions', () => {
       expect(responses.length).toBe(2)
       expect(responses[1].status).toBe(201)
     })
+    it('correctly maps custom variables', async () => {
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/__tests__/uploadClickConversion.test.ts:149]: should we give a different name to this test? Like `uses v12 when google-enhanced-v12 flag is enabled`?

File: packages/destination-actions/src/destinations/google-enhanced-conversions/types.ts [hunk 3]
```diff
@@ -24,4 +26,9 @@ export interface PartialErrorResponse {
   results: {}[]
 }
 
-export const GoogleAdsAPI = 'https://googleads.googleapis.com/v11/customers'
+export function getUrlByVersion(features: Features | undefined): string {
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/types.ts:29]: Can we add the link to the ticket for removing this flagon in a comment above this method?

**Review by varadarajan-tw — COMMENTED** [review]
Looks good! 🔥  Just a few minor comments i'd like to see addressed.
Also can you update the PR description with the correct production flagon gate url? Seems to be pointing to the old one.

**Review by marinhero — COMMENTED** [review]
Left a small comment, it's a nit so feel free to disregard it.

**Review by varadarajan-tw — APPROVED** [review]
Thanks for making the recommended changes. Pls wait till the deployment freeze is lifted before merging.

**Pull request merged**